indexer/src/main/scala/scalding/BaseFeaturesImporterJob.scala

```diff
@@ -0,0 +1,191 @@
+// Copyright 2014 Foursquare Labs Inc. All Rights Reserved.
+package com.foursquare.twofishes.scalding
+
+import com.twitter.scalding._
+import org.apache.hadoop.io.LongWritable
+import com.twitter.scalding.typed.TypedSink
+import com.foursquare.twofishes._
+import com.foursquare.hadoop.scalding.SpindleSequenceFileSource
+import com.foursquare.twofishes.importers.geonames.GeonamesFeature
+import com.foursquare.twofishes.util._
+import com.vividsolutions.jts.io.{WKBWriter, WKTReader}
+
+class BaseFeaturesImporterJob(
+  name: String,
+  lineProcessor: (Int, String) => Option[GeonamesFeature],
+  allowBuildings: Boolean = false,
+  inputSpec: TwofishesImporterInputSpec,
+  args: Args
+) extends TwofishesImporterJob(name, inputSpec, args) {
+
+  private def getDisplayNamesFromGeonamesFeature(feature: GeonamesFeature): List[DisplayName] = {
+    def isAllDigits(x: String) = x.forall(Character.isDigit)
+
+    // add primary name as English PREFERRED
+    List(DisplayName("en", feature.name, FeatureNameFlags.PREFERRED.getValue)) ++
+    // add ascii name as English DEACCENTED
+    feature.asciiname.map(n => DisplayName("en", n, FeatureNameFlags.DEACCENT.getValue)).toList ++
+    // add country code as abbreviation if this is a country
+    (if (feature.featureClass.woeType.getValue == YahooWoeType.COUNTRY.getValue) {
+      List(DisplayName("abbr", feature.countryCode, 0))
+    } else {
+      Nil
+    }) ++
+    // the admincode is the internal geonames admin code, but is very often the
+    // same short name for the admin area that is actually used in the country
```

Comment:
Sounds like we can actually use this pretty aggressively, like Rome => RM is correct